cdp-agentkit-core/typescript/src/action_providers/farcaster/farcasterActionProvider.ts

```diff
@@ -0,0 +1,155 @@
+import { z } from "zod";
+import { ActionProvider } from "../action_provider";
+import { Network } from "../../wallet_providers";
+import { CreateAction } from "../action_decorator";
+import { AccountDetailsSchema, PostCastSchema } from "./schemas";
+
+/**
+ * Configuration options for the FarcasterActionProvider.
+ */
+export interface FarcasterActionProviderConfig {
+  /**
+   * Neynar API Key.
+   */
+  neynarApiKey?: string;
+
+  /**
+   * Neynar managed signer UUID.
+   */
+  signerUuid?: string;
+
+  /**
+   * Agent FID.
+   */
+  agentFid?: string;
+}
+
+/**
+ * FarcasterActionProvider is an action provider for Farcaster.
+ */
+export class FarcasterActionProvider extends ActionProvider {
+  private readonly neynarApiKey: string;
+  private readonly signerUuid: string;
+  private readonly agentFid: string;
+
+  constructor(config: FarcasterActionProviderConfig = {}) {
+    super("farcaster", []);
+
+    if (config.neynarApiKey) {
+      this.neynarApiKey = config.neynarApiKey;
+    } else if (process.env.NEYNAR_API_KEY) {
+      this.neynarApiKey = process.env.NEYNAR_API_KEY;
+    } else {
+      throw new Error("NEYNAR API Key is not configured.");
+    }
+
+    if (config.signerUuid) {
+      this.signerUuid = config.signerUuid;
+    } else if (process.env.NEYNAR_MANAGER_SIGNER) {
+      this.signerUuid = process.env.NEYNAR_MANAGER_SIGNER;
+    } else {
+      throw new Error("NEYNAR Managed Signer UUID is not configured.");
+    }
+
+    if (config.agentFid) {
+      this.agentFid = config.agentFid;
+    } else if (process.env.AGENT_FID) {
+      this.agentFid = process.env.AGENT_FID;
+    } else {
+      throw new Error("Agent FID is not configured.");
+    }
+  }
+
+  /**
+   * Checks if the Farcaster action provider supports the given network.
+   *
+   * @param network - The network to check.
+   * @returns True if the Farcaster action provider supports the network, false otherwise.
+   */
+  supportsNetwork = (_: Network) => true;
+
+  /**
+   * Retrieves agent's Farcaster account details.
+   *
+   * @param _ The input arguments for the action.
+   * @returns A message containing account details for the agent's Farcaster account.
+   */
+  @CreateAction({
+    name: "account_details",
```

Comment:
should the action names be camelCased too?

Reply:
I would prefer for those to be the same across languages. Langchain does this as well